test/parallel/test-fs-realpath.js

```diff
@@ -201,9 +201,11 @@ function test_cyclic_link_protection(callback) {
     fs.symlinkSync(t[1], t[0], 'dir');
     unlink.push(t[0]);
   });
-  assert.throws(function() { fs.realpathSync(entry); });
+  assert.throws(function() { fs.realpathSync(entry); }, /^Error: ELOOP/);
   asynctest(fs.realpath, [entry], callback, function(err, result) {
```

Comment:
It would likely be worthwhile to wrap the `function(err, result) {}` block in a `common.mustCall()` throughout as an extra check